### DISCUSSION TITLE
Could not query netbox data when sync via netbox-sync

### DISCUSSION THREAD

**Discussion by ngocnd1** [netbox-community/netbox#12003] [Q&A]
Hi all,

Does anyone get same issues, with version netbox 3.4.6,
When I synced via netbox-sync it could not query from netbox API due to contextlimit as below,
Is there anyone has experience with this issue, i try with netbox 3.4.5 it's still working fine, Manythanks
![Capture](https://user-images.githubusercontent.com/64430144/225954995-fe70af1a-a7c6-4908-b8d3-ea3bf8609a23.PNG)

**Comment by candlerb** [comment]
I suggest you use a tcpdump:

```
tcpdump -i lo -nn -s0 -A tcp port 8001
```

to see the query sent to Netbox, and more importantly, the error response which is returned.

It works for me (v3.4.6) - even when the offset is greater than the number of VMs:

```
# curl -H "Authorization: Token $TOKEN" 'http://127.0.0.1:8001/api/virtualization/virtual-machines/?exclude=config_context&last_updated__gte=2023-03-17T17%3A06%3A15.837532%2B07%3A00&limit=200&offset=200'
{"count":0,"next":null,"previous":"http://127.0.0.1:8001/api/virtualization/virtual-machines/?exclude=config_context&last_updated__gte=2023-03-17T17%3A06%3A15.837532%2B07%3A00&limit=200","results":[]}
```

Therefore, either you've provided the wrong credentials:

```
{"detail":"Authentication credentials were not provided."}
```

although that doesn't explain why the *first* 200 VMs were presumably fetched successfully.

Or you've hit some other case to do with results paging, in which case we need to see the error message returned from Netbox to the API call.

**Reply by ngocnd1** [reply]
Firstly thanks for your help
Could i ask more, Where do i need to run tcp dump? netbox server or netbox-sync server?
Actually i have a test, when i delete data on netbox, from 260 vms -> 199 vms so It's is working fine :D

**Reply by candlerb** [reply]
> Could i ask more, Where do i need to run tcp dump? netbox server or netbox-sync server?

They are clearly the same thing, since netbox-sync is talking to netbox on 127.0.0.1 (localhost).

> Actually i have a test, when i delete data on netbox, from 260 vms -> 199 vms so It's is working fine :D

Then it seems there's some problem with either netbox or netbox-sync when paging to the second page of data.  If you could capture the tcpdump, that would allow it to be fixed.

**Reply by ngocnd1** [reply]
when i ran tcpdump on netbox-sync server result as below
listening on lo, link-type EN10MB (Ethernet), captrue size 262144 bytes
Do you have any recommend?

**Reply by ngocnd1** [reply]
![Capture1](https://user-images.githubusercontent.com/64430144/225981066-1b5f244e-2c6e-4f23-a6f3-042250f9ef0e.PNG)

**Reply by candlerb** [reply]
This shows that Netbox isn't running, on the host where netbox-sync is running.

Sorry, but I think you need to go back to basics:
1. identify which host Netbox is running on, and make sure it is running
2. identify which host netbox-sync is running on
3. configure netbox-sync to point to the correct host from (1) - because clearly either 127.0.0.1 is wrong, or Netbox isn't running  at all.

**Reply by ngocnd1** [reply]
Thanks for your instrucstion.
